### DISCUSSION TITLE
Is it possible to validate a plain JSON Schema?

### DISCUSSION THREAD

**Discussion by silwol** [stoplightio/spectral#2653] [Q&A]
Validating an OpenAPI specification file is as simple as putting `extends: "spectral:oas"` into a `.yaml` file, and passing this in as a parameter when calling `spectral lint`.

I have a use case where I don't have an OpenAPI specification inside a file, but just a plain JSON schema which I'd check for consistency? Is that possible? Given the fact that the `spectral:oas` ruleset relies on JSON Schema, and the `draft-2020-12` JSON schema definition is even included in the source tree, this should be rather straight-forward, right? Is it exposed as functionality somewhere? I tried using `spectral:jsonschema` and `spectral:json-schema`, but none of these worked.

**Comment by uuf6429** [comment]
I have this same problem... I spent some time setting up spectral only to learn that by default it won't do anything. I was kind of expecting some basic rules like "description must be set in specific cases", "object properties must be camelCased" or some complex checks to rule out impossible situations like `allOf: [string, object]` etc.

**Comment by erwinkramer** [comment]
I just built this custom function, it's a bit of a poor man's implementation of JSON schema validation, but it does just what i need it to do: https://github.com/erwinkramer/bank-api/blob/main/.spectral/functions/well-formed-object.js

**Comment by damnthonyy** [comment]
I've created [json-schema-validate](https://github.com/damnthonyy/json-schema-validate); a package that combines swagger-parser (for structural validation + circular detection) and Spectral (for quality rules) to properly handle what both tools alone can't do.

``` 
npm install @damnthonyy/json-schema-validate
```
```
import { validateSchema } from '@damnthonyy/json-schema-validate';

const result = await validateSchema(spec);

if (result.valid) {
  console.log('✅ Spec is valid');
} else {
  result.issues.forEach(issue => {
    console.log(`[${issue.severity}] ${issue.source}: ${issue.message}`);
  });
}
```

## What it validates
- [x] Broken $ref references
- [x] Circular references (Pet → Owner → Pet)
- [x] Invalid OpenAPI format
- [x] Quality issues (empty descriptions, missing types, OAS compliance)
